test/GitcoinGovernor.t.sol

```diff
@@ -2,13 +2,14 @@
 pragma solidity ^0.8.17;
 
 import "forge-std/Test.sol";
+import {IERC20} from "openzeppelin-contracts/interfaces/IERC20.sol";
```

Comment:
_(arbitrary comment location to start a thread)_

Not exactly related to this PR, but in `setUp` we call `vm.createSelectFork(vm.rpcUrl("mainnet"), _forkBlock);`. However, for these tests it's pretty important to make sure they pass against the latest state, and not a pinned fork block. Some options:
1. Remove the pinned block and always use latest—downside is this slows down tests locally
2. Change the fork setup to something like below to only fork from pinned block locally, and use latest in CI

```solidity
if (keccak256(vm.envString("FOUNDRY_PROFILE")) == keccak256("ci")) {
  vm.createSelectFork(vm.rpcUrl("mainnet")); // Use latest block.
} else [
  vm.createSelectFork(vm.rpcUrl("mainnet"), _forkBlock); // Use pinned block.
}
```

One issue is this might break your "prank as `delegates` to get proposal to pass if their votes have changed, but in that case you can either (1) change from pranking as delegates to just overwriting storage slots, or (2) add new delegates to the list / have two lists

Reply:
> However, for these tests it's pretty important to make sure they pass against the latest state, and not a pinned fork block

Can you expand a bit on what you mean here? I don't think it's actually too big of a deal to be a bit behind on the state. We're still exercising everything that matters and there's not much that could/would change which would impact it.

For what it's worth, I think we will deploy the new Governor and update tests to use a later block and the already-deployed Governor after that.

Also I tried to write the tests as agnostic to state as possible so we can (and should) update to a later block shortly before submitting the proposal and ensure everything still passes. As you mentioned one thing that could change is delegates, but I'd be surprised if they changed too dramatically too quickly.

All this to say, I don't think it's critical to *always* run the tests against the latest state in the meantime.

Reply:
> For what it's worth, I think we will deploy the new Governor and update tests to use a later block and the already-deployed Governor after that.

This seems like a good alternative

> Can you expand a bit on what you mean here? I don't think it's actually too big of a deal to be a bit behind on the state. We're still exercising everything that matters and there's not much that could/would change which would impact it.
>
> All this to say, I don't think it's critical to always run the tests against the latest state in the meantime.

Basically, it's cheap to run tests against the latest block, and doing so provides more confidence than a block that's 1-3 months old. I can't give a good answer for what specifically might change that could cause a future failure, but feel like I've seen enough weird things happen + it's cheap enough to do + upgrades are high stakes, that it feels worth being more conservative

Reply:
Yeah fair points! I will update to later block in the next PR to keep us relatively recent, and also update the block again post Governor-deploy, and before the proposal, as mentioned above.